### ISSUE TITLE
web build not fine

### ISSUE THREAD

**Issue by krmbn0576** [EasyRPG/buildscripts#124]
web(emscripten) build in WSL1 not fine?

```
$ git clean -xdf && ./0_build_everything.sh

**** Building json ****
configure: cmake . -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_SHARED_LIBS=OFF "-DCMAKE_C_FLAGS=-O2 -g0 -I/mnt/c/Users/-/Documents/EasyRPG/buildscripts/emscripten/include" "-DCMAKE_CXX_FLAGS=-O2 -g0 -I/mnt/c/Users/-/Documents/EasyRPG/buildscripts/emscripten/include" -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_INSTALL_PREFIX=/mnt/c/Users/-/Documents/EasyRPG/buildscripts/emscripten -DCMAKE_SYSTEM_NAME= -DJSON_BuildTests=OFF -DCMAKE_TOOLCHAIN_FILE=/mnt/c/Users/-/Documents/EasyRPG/buildscripts/emscripten/emsdk-portable/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake -DCMAKE_CROSSCOMPILING_EMULATOR="/mnt/c/Users/-/Documents/EasyRPG/buildscripts/emscripten/emsdk-portable/node/14.15.5_64bit/bin/node"
-- Using the single-header code from /mnt/c/Users/-/Documents/EasyRPG/buildscripts/emscripten/json-3.10.4/single_include/
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_C_FLAGS


-- Build files have been written to: /mnt/c/Users/-/Documents/EasyRPG/buildscripts/emscripten/json-3.10.4
make[1]: *** ターゲット 'clean' を make するルールがありません.  中止.
Makefile:136: recipe for target 'clean' failed
make: *** [clean] Error 2
```

hmm `nlohmann-json` ...

**Comment by Ghabry** [comment]
nlohmann is not needed by the Player. You can delete that line in the 2.sh and try again.

Unfortunately hard to test fort me. Only have WSL2

**Comment by krmbn0576** [comment]
thx 😇 

but another error

```
../src/player.cpp:1216:41: error: no matching function for call to 'DetectEncodings'
                        std::vector<std::string> encodings = lcf::ReaderUtil::DetectEncodings(ldb_stream);
                                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../lib/liblcf/src/lcf/reader_util.h:51:27: note: candidate function not viable: no known conversion from 'Filesystem_Stream::InputStream' to 'lcf::rpg::Database &' for 1st argument
        std::vector<std::string> DetectEncodings(lcf::rpg::Database& db);
                                 ^
../lib/liblcf/src/lcf/reader_util.h:70:27: note: candidate function not viable: no known conversion from 'Filesystem_Stream::InputStream' to 'lcf::StringView' (aka 'basic_string_view<char>') for 1st argument
        std::vector<std::string> DetectEncodings(StringView string);
                                 ^
1 error generated.
```

🤔

**Comment by Ghabry** [comment]
Your liblcf is too old. Two solutions:

First option:

Go into [PLAYER]/lib/liblcf folder and run "git pull". Then use "cmake --build" as usual.

Second option:

Delete lib/liblcf folder and run that long CMake command line again and build.

**Comment by krmbn0576** [comment]
First result:

```
/mnt/c/Users/-/Documents/EasyRPG/Player/lib/liblcf (master)$ git log -t
commit 01b73de93cf80185fcdf3ffd4737dfdb9111a85e (grafted, HEAD -> master, tag: 0.7.0, origin/master, origin/HEAD)
Author: Carsten Teibes <dev@f4ke.de>
Date:   Fri Oct 29 18:42:43 2021 +0200

    Merge pull request #437 from Ghabry/release/0.7.0

    Release 0.7.0
/mnt/c/Users/-/Documents/EasyRPG/Player/lib/liblcf (master)$ git pull
Already up to date.
```

Second option:

```
$ rm -rf lib
$ cmake . -GNinja -Bembuild -DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=BOTH -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=BOTH -DCMAKE_FIND_ROOT_PATH_MODE_PACKAGE=BOTH -DPLAYER_BUILD_LIBLCF=ON -DLIBLCF_WITH_XML=OFF -DLIBLCF_ENABLE_TOOLS=OFF -DCMAKE_TOOLCHAIN_FILE=$PATH_TO_BUILDSCRIPTS/emscripten/emsdk-portable/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake -DCMAKE_PREFIX_PATH=$PATH_TO_BUILDSCRIPTS/emscripten -DCMAKE_BUILD_TYPE=Release -DPLAYER_JS_BUILD_SHELL=ON -DPLAYER_JS_OUTPUT_NAME=index -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
$ cmake --build embuild

(same error)
```

**Comment by Ghabry** [comment]
Sorry I misread the error message.

Your Player code is too old 😅.

Run in the Player directory:

```
git remote add upstream https://github.com/easyrpg/player
git fetch upstream
git rebase upstream/master # This can give conflicts you must resolve
```

Now your code is based on the latest Player code again

**Comment by krmbn0576** [comment]
```
$ git log -t
commit 5e8f0a4efe918364de6823972624268f28d40bf6 (HEAD -> master, origin/master, origin/HEAD)
Merge: 33883b15 e1e52dc8
Author: Gabriel <gabriel+github@mastergk.de>
Date:   Fri Nov 5 15:58:35 2021 +0100

    Merge pull request #2684 from rueter37/issue-2679

    Set frame_limit to zero if fps_limit is set to zero

$ git remote -v
atsumaru        https://github.com/atsumaru/Player.git (fetch)
atsumaru        https://github.com/atsumaru/Player.git (push)
origin  https://github.com/EasyRPG/Player.git (fetch)
origin  https://github.com/EasyRPG/Player.git (push)

$ git pull origin master
From https://github.com/EasyRPG/Player
 * branch              master     -> FETCH_HEAD
Already up to date.
```

Ohh... 🤔

**Comment by Ghabry** [comment]
Ah in that case you must be on the atsumaru branch and run ``git rebase origin/master``.

Somehow you are building old code this "DetectEncoding" was changed a few months ago

**Comment by krmbn0576** [comment]
```
$ git rebase origin/master
Current branch master is up to date.
```

Hmm I also looked at `player.cpp` local and remote repositories directly, and there is no noticeable change in the DetectEncodings call

**Comment by krmbn0576** [comment]
To avoid confusion, I haven't checked out the `atsumaru` repository now. Only use origin

**Comment by Ghabry** [comment]
Sorry I am clueless.

The ``DetectEncodings`` line should be:

```cpp
std::vector<std::string> encodings = lcf::ReaderUtil::DetectEncodings(*db);
```

as shown here https://github.com/EasyRPG/Player/blob/master/src/player.cpp#L1226

Somehow you did not get that change but I have no idea why :thinking:

**Comment by krmbn0576** [comment]
>git reset --hard origin/master

Thank you very much! This command worked and the problem has been solved!

The problem was in git and in `src/docmain.h`. Only this file uses `CRLF`, which causes line feed discrepancies in the Windows file system, causing pull and checkout of the latest code to fail. If this file is changed to `LF`, there seems to be no problem.

**Comment by Ghabry** [comment]
ooooh, this problem.

Yeah we had the same issue! When this file is not changed git becomes unusable.

We fixed this in 0.7.0 so you shouldn't see that error again.
